daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -124,72 +124,85 @@ func (l *JSONFileLogger) Log(msg *logger.Message) error {
 }
 
 func writeLog(l *JSONFileLogger) (int64, error) {
+
+	if err := checkCapacityAndRotate(l); err != nil {
+		return -1, err
+	}
+
+	i, err := l.buf.WriteTo(l.f)
+	if err != nil {
+		l.buf.Reset()
```

Comment:
oooh, sorry, I misread the code. I thought you were calling `Reset()` if there was no error.
Doing it this way defeats the point of the buffer. The purpose of the buffer in the driver (as far as I can tell) is so that if the write fails, the data won't be lost, and will try to be written again next time something comes into the driver.
With this change, the buffer no longer has a purpose. On write success the buffer is flushed, and on write fail the buffer is flushed.